R/WebApi.R

```diff
@@ -142,3 +142,39 @@ getCdmSources <- function(baseUrl) {
 
   do.call(rbind, sourceDetails)
 }
+
+
+#' Get the WebApi configuration for the WebApi instance.
+#'
+#' @details
+#' Get the WebApi configuration for the WebApi instance.
+#'
+#' @param baseUrl        The base URL for the WebApi instance, for example:
+#'                       "http://server.org:80/WebAPI".
+#' @param sourceKeys     A list of string names for the source keys as in
+#'                       configuration page of Atlas.
+#' @return
+#' A list representing the WebApi configuration for the CDM Schema
+#'
+#' @examples
+#' \dontrun{
+#' #
+#'
+#' getConfiguration(baseUrl = "http://server.org:80/WebAPI", sourceKeys = c('source1', 'source2'))
+#' }
+#'
+#' @export
+getConfiguration <- function(baseUrl,sourceKeys){
+  require(dplyr)
+  .checkWebApiBaseUrl(baseUrl)
+  
+  a <- .getCdmSources(baseUrl)
+  a$parsed <- a$parsed %>%
```

Comment:
I'm pretty sure the `%>%` notation requires the dplyr package to be loaded on the search path, which we don't want to do. Use :: notation instead.

Reply:
Just make sure the function works without requiring the user to load magitr on their search path (in other words, without requiring the user to type `library('magitr')` before running your function)

Reply:
I dont know the answer. if dplyr is being used in the package, then dplyr is supposed to re-export magitr and attach it. dplyr is required for the package. 

i will do some research.

Reply:
@schuemie I think we can address this by adding magitr to 'Depends:' section of DESCRIPTION file. 
I made the changes, and loaded the package - and found that magitr was attached.

![image](https://user-images.githubusercontent.com/13936600/72072914-df4fa280-32bc-11ea-952e-2e8eab656d33.png)

Reply:
Yes, I know that would solve it ;-) But I'm explicitly saying that is not an option: we want to minimize the side-effects of running a package and running a function, meaning we should avoid changing the user's search path if possible. If you look at the other OHDSI R packages, the only things that you may find in the Depends section are DatabaseConnector and sometimes Cyclops, because the user must have those on the search path to use the functions in the package. For example, the use must use createConnectionDetails to use CohortMethod. Anything that is not absolutely required to use the package should be in the Imports section, meaning we can use it in the package, but it does not get loaded on the search path. That means we need to use the :: to reference the package functions, which gets a bit tricky for magrittr's weird operators. It is still possible, but you should write `magrittr::`%>%`(a, b)` instead of `a  %>% b`.

Reply:
thanks martijn! we will follow this

Reply:
@schuemie using `magrittr::%>%(a,b)` would make this practically not usable. 

I dont think it is reasonable to translate an easy to follow code like below to the form `magrittr::%>%(a,b)` .
```

    daimons <- result$parsed$daimons
	daimons
      dplyr::bind_rows() %>%
      tidyr::spread(key = daimonType, value = tableQualifier) %>%
      dplyr::select(-sourceDaimonId) %>%
      base::replace(is.na(.), "") %>%
      dplyr::group_by(sourceId) %>%
      dplyr::summarise_all(max) %>%
      dplyr::mutate_all(dplyr::na_if,"") %>% 
      dplyr::rename(
        cdmDatabaseSchema = CDM, #case sensitve
        vocabDatabaseSchema = Vocabulary,
        resultsDatabaseSchema = Results
      )
```

An alternative would be to convert to 
```

  daimons <- result$parsed$daimons
  daimons <- dplyr::bind_rows(daimons)
  daimons <- tidyr::spread(daimons, key = daimonType, value = tableQualifier)
  daimons <- dplyr::select(daimons, -sourceDaimonId)
  daimons[is.na(daimons)] <- ""
  daimons <- dplyr::group_by(daimons, sourceId)
  daimons <- dplyr::summarise_all(daimons, max)
  daimons <- dplyr::mutate_all(daimons, dplyr::na_if, "")
```

@schuemie because magrittr is such a useful package, i would like to propose that we add it to 'Depends' in the interest of code readability and ease of use. Infact, it would be great to add magrittr to the OHDSI R packages. in general..

Reply:
This is a good read

https://community.rstudio.com/t/magrittr-inside-a-package/2033

Reply:
Easiest solution seems to be to import the magrittr inflix operator for every function
```
#' @importFrom magrittr "%>%"
#'
```

There is another option with `usethis`
https://usethis.r-lib.org/reference/use_pipe.html